Packs/CommonScripts/Scripts/IsDomainInternal/IsDomainInternal.py

```diff
@@ -0,0 +1,93 @@
+from time import sleep
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+POLLING_TIME = 2  # Interval to wait in seconds when polling to check if indicator was created
+
+
+def check_domain_internal(domain, internal_domains):
+    parts = domain.split(".")
+    for i in range(len(parts), 0, -1):
+        sub = ".".join(parts[-i:])
+        if sub in internal_domains:
+            return True
+    return False
+
+
+def create_markdown_table(domain_list):
+    domain_list = sorted(domain_list, key=lambda x: not x['Internal'])
+    markdown_string = "| Domain Name | Internal |\n| --- | --- |\n"
+    for domain in domain_list:
+        markdown_string += f"| {domain['Name']} | {domain['Internal']} |\n"
+    return markdown_string
+
+
+internal_domains_list = demisto.args().get("InternalDomainsListName", "InternalDomains")
+domains_to_check = argToList(demisto.args().get("Domains", None))
+
+# Get the list of internal domains from the XSOAR list:
+internal_domains = demisto.executeCommand("getList", {"listName": internal_domains_list})[0]['Contents']
+if "Item not found" in internal_domains:
+    return_error(f"The list name {internal_domains_list} does not exist.")
+
+# Split internal domains from XSOAR list to be a list:
+if internal_domains:
+    try:
+        internal_domains = internal_domains.split("\n")
+    except Exception as ex:
+        return_error(f"Could not parse the internal domains list. Please make sure that the list contains domain names, separated by new lines.\nThe exact error is: {ex}")
+else:
+    demisto.results("No internal domains were specified.")
+    quit()
+
+
+# Create list of domain names with internal property
+domain_list = [{"Name": domain, "Internal": check_domain_internal(domain, internal_domains)} for domain in domains_to_check]
+
+
+for domain in domain_list:
+    args_exists_check = {
+        "indicator": domain.get("Name")
+    }
+
+    args_create_or_set_indicator = {
+        "value": domain.get("Name"),
+        "type": "Domain",
+        "internal": domain.get("Internal")
+    }
+
+    # Check if indicator exists:
+    is_exist_res = demisto.executeCommand("CheckIndicatorValue", args_exists_check)
+    indicator_exists = is_exist_res[0].get("Contents", {})[0].get("Exists")
+
+    # If indicator doesn't exist, create it and continuously poll for its creation (which happens asynchronously):
+    if not indicator_exists:
+        demisto.executeCommand("createNewIndicator", args_create_or_set_indicator)
+        while not indicator_exists:  # Looping because it takes time for the indicator to be created
+            is_exist_res = demisto.executeCommand("CheckIndicatorValue", args_exists_check)
+            indicator_exists = is_exist_res[0].get("Contents", {})[0].get("Exists")
+            sleep(POLLING_TIME)
```

Comment:
can we use a [scheduled (polling) command](https://xsoar.pan.dev/docs/integrations/scheduled-commands) instead? `sleep` is not a good practice in XSOAR due to the container architecture.

Reply:
Yeah I figured sleep would be bad, I will take a look at the polling command tomorrow

Reply:
After discussing with you + Binat, seems like it's meant for integration commands and not scripts. Let's keep sleep at this point